sources/dogtag/bin/imds.rs

```diff
@@ -0,0 +1,39 @@
+use dogtag::Cli;
+use imdsclient::ImdsClient;
+use snafu::{OptionExt, ResultExt};
+
+type Result<T> = std::result::Result<T, error::Error>;
+
+/// Implements a hostname lookup tool by fetching the public hostname
+/// from the instance metadata via IMDS. It will interface with IMDS
+/// via:
+///
+/// * Check for IPv6, default to IPv4 if not available
+/// * Check for IMDSv2, fallback to IMDSv1 if not enabled
```

Comment:
I don't think these comments apply; `imdsclient` always uses IMDSv2 and the IPv4 address (which is still valid on IPV6-only ENIs).